agent/rpc/peering/stream_test.go

```diff
@@ -63,10 +66,32 @@ func TestStreamResources_Server_Follower(t *testing.T) {
 		}
 	}()
 
+	// expect error
 	msg, err := client.Recv()
 	require.Nil(t, msg)
 	require.Error(t, err)
 	require.EqualError(t, err, "rpc error: code = FailedPrecondition desc = cannot establish a peering stream on a follower node")
+
+	// expect a status error
+	st, ok := status.FromError(err)
+	require.True(t, ok, "need to get back a grpc status error")
+	deets := st.Details()
+	// expect a ReplicationMessage_LeaderAddress
+	{
+		found := false
+		for _, deet := range deets {
```

Comment:
this is pretty good approximation of how the dialer will look for a leader address if we don't tag the server addresses